### PR TITLE
Extend SDL logs

### PR DESCRIPTION
**Pull request by anosach-luxoft**
- Add line number, file name and function name to log string
- Add auto trace logs

### CODE REVIEW

**Comment by anosach-luxoft** [comment]
@malirod, @nk0leg, @pestOO please review

File: src/components/include/utils/logger.h [hunk 1]
```diff
@@ -140,36 +140,66 @@ namespace logger {
...
+     accumulator << __FILE__ << ":" << line << " " << __FUNCTION__ << ": " << logEvent; \
+     logger::push_log(loggerPtr, logLevel, logger::time_now(), accumulator.str()); \
+} while (false)
 
-#define LOG_WITH_LEVEL(loggerPtr, logLevel, logEvent) \
+#define LOG_WITH_LEVEL_EXT(loggerPtr, logLevel, logEvent, line, file, function) \
 do { \
      std::stringstream accumulator; \
-     accumulator << logEvent; \
+     accumulator << file << ":" << line << " " << function << ": " << logEvent; \
```
Comment (pestOO) [inline src/components/include/utils/logger.h]: It is redundant, Log4cxx already has this!

File: src/components/include/utils/logger.h [hunk 2]
```diff
@@ -140,36 +140,66 @@ namespace logger {
...
      logger::push_log(loggerPtr, logLevel, logger::time_now(), accumulator.str()); \
 } while (false)
 
+namespace logger {
+  class AutoTrace {
```
Comment (pestOO) [inline src/components/include/utils/logger.h]: What with already exist `AutoTrace`?

File: src/components/include/utils/logger.h [hunk 3]
```diff
@@ -140,36 +140,66 @@ namespace logger {
...
 
-#define LOG4CXX_AUTO_TRACE_WITH_NAME_SPECIFIED(loggerPtr, auto_trace)// \
-    //logger::AutoTrace auto_trace(loggerPtr, LOG4CXX_LOCATION)
-#define LOG4CXX_AUTO_TRACE(loggerPtr) //LOG4CXX_AUTO_TRACE_WITH_NAME_SPECIFIED(loggerPtr, SDL_local_auto_trace_object)
+#define LOG4CXX_AUTO_TRACE(loggerPtr) \
+    logger::AutoTrace auto_trace(loggerPtr, __LINE__, __FILE__, __FUNCTION__);
 
 #undef LOG4CXX_DEBUG
-#define LOG4CXX_DEBUG(loggerPtr, logEvent) LOG_WITH_LEVEL(loggerPtr, 1, logEvent)
+#define LOG4CXX_DEBUG(loggerPtr, logEvent) LOG_WITH_LEVEL(loggerPtr, 1, logEvent, __LINE__)
```
Comment (pestOO) [inline src/components/include/utils/logger.h]: 0,1 and so on is not good enough logging levels.
It's not clear what is the top and bottom)

File: src/components/include/utils/logger.h (hunk 1 above)
Comment (malirod) [inline src/components/include/utils/logger.h]: In windows & qt versions we have no log4cxx. And this header is shared among all platforms. And this macro is used on win platform.

File: src/components/include/utils/logger.h (hunk 2 above)
Comment (anosach-luxoft) [inline src/components/include/utils/logger.h]: @pestOO, existing `AutoTrace` uses `log4cxx`, so we cann't use it

File: src/components/include/utils/logger.h (hunk 3 above)
Comment (anosach-luxoft) [inline src/components/include/utils/logger.h]: @pestOO, I agree it's not clearest solution, but we use this way in linux version. What do you propose?

Comment (pestOO) [inline src/components/include/utils/logger.h]: Add enum with LogLevel names at least.

**Comment by anosach-luxoft** [comment]
@pestOO, @malirod, @nk0leg pull request has been updated

**Comment by malirod** [comment]
Reviewed

**Comment by nk0leg** [comment]
Reviewed

**Pull request merged**